app/src/main/java/fr/gaulupeau/apps/Poche/service/EventProcessor.java

```diff
@@ -0,0 +1,283 @@
+
+    private static final int NOTIFICATION_ID_UPDATE_FEEDS_ONGOING = 1;
+
+    private Context context;
+    private Settings settings;
+    private Handler mainHandler;
+    private NotificationManager notificationManager;
+
+    private Long currentOperationID; // TODO: replace with ActionRequest?
+
+    private boolean delayedNetworkChangedTask;
+
+    public EventProcessor(Context context) {
+        this.context = context;
+    }
+
+    public void start() {
+        EventBus.getDefault().register(this);
+    }
+
+    public void stop() {
+        EventBus.getDefault().unregister(this);
+    }
+
+    @Subscribe // TODO: check thread
+    public void onConnectivityChangedEvent(ConnectivityChangedEvent event) {
+        Log.d(TAG, "onConnectivityChangedEvent() started");
+
+        networkChanged(false);
+    }
+
+    @Subscribe
+    public void onOfflineQueueChangedEvent(OfflineQueueChangedEvent event) {
+        Log.d(TAG, "onOfflineQueueChangedEvent() started");
+
+        Long queueLength = event.getQueueLength();
+
+        Log.d(TAG, "onOfflineQueueChangedEvent() offline queue length: " + queueLength);
+
+        getSettings().setBoolean(Settings.PENDING_OFFLINE_QUEUE,
+                queueLength == null || queueLength != 0);
+    }
+
+    @Subscribe(sticky = true)
+    public void onUpdateFeedsStartedEvent(UpdateFeedsStartedEvent event) {
+        Log.d(TAG, "onUpdateFeedsStartedEvent() started");
+
+        setOperationID(event);
+
+        FeedUpdater.FeedType feedType = event.getFeedType();
+
+        // TODO: fix notification content
+        String detailedMessage;
+        if(feedType == null) detailedMessage = "Updating all feeds";
+        else detailedMessage = String.format("Updating %s feed", feedType);
+        NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(getContext())
+                .setSmallIcon(R.drawable.ic_action_refresh)
+                .setContentTitle("Updating feeds")
+                .setContentText(detailedMessage)
+                .setOngoing(true);
+
+        getNotificationManager().notify(TAG, NOTIFICATION_ID_UPDATE_FEEDS_ONGOING,
+                notificationBuilder.setProgress(0, 0, true).build());
+    }
+
+    @Subscribe
+    public void onUpdateFeedsFinishedEvent(UpdateFeedsFinishedEvent event) {
+        Log.d(TAG, "onUpdateFeedsFinishedEvent() started");
+
+        checkOperationID(event);
+
+        getNotificationManager().cancel(TAG, NOTIFICATION_ID_UPDATE_FEEDS_ONGOING);
+
+        emptyOperationID();
+    }
+
+    @Subscribe
+    public void onActionResultEvent(ActionResultEvent event) {
+        Log.d(TAG, "onActionResultEvent() started");
+
+        Notification notification = null;
+
+        ActionRequest request = event.getRequest();
+        String actionString = request.getAction().toString();
+
+        ActionResult result = event.getResult();
+
+        // TODO: fix: debug only implementation
+
```

Comment:
Maybe a notification & a toast each time you do something as basic as archiving/favoriting is too much ?